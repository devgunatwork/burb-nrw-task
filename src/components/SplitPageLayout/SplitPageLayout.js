import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@material-ui/core/Grid';
import styles from './SplitPageLayout.module.scss';

/**
 * @description HOC to split page into fixed LHS with image & scrollable RHS with dynamic content
 * @param children: optional content passed
 * @param imageUrl: url for fixed image on RHS
 * */
const SplitPageLayout = ({ children, imageUrl }) => {
  return (
    <Grid item xs={12}>
      <Grid container className={styles.root}>
        <Grid item xs={12} sm={12} md={6} lg={6} xl={4} className={styles.leftSide}>
          <img alt="not available" src={imageUrl} />
        </Grid>
        <Grid item xs={12} sm={12} md={6} lg={6} xl={8} className={styles.rightSide}>
          {children}
        </Grid>
      </Grid>
    </Grid>
  );
};

SplitPageLayout.defaultProps = {
  children: '',
  imageUrl: '',
};
SplitPageLayout.propTypes = {
  children: PropTypes.node,
  imageUrl: PropTypes.string,
};

export default SplitPageLayout;
