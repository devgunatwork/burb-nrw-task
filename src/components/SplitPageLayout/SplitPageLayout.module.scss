@import "src/style/mixins/mixins";

.root {
  position: relative;
  min-width: 360px;

  .leftSide {
    bottom: 0;
    left: 0;
    position: fixed;
    top: 0;

    img {
      height: 100%;
      width: 100%;
      object-fit: cover;
    }
  }

  .rightSide {
    padding: 1em 2em;
    position: absolute;
    max-height: 100vh;
    right: 0;
  }

  @include media(screen-xs, screen-sm) {
    .leftSide, .rightSide {
      position: relative;
    }
  }
}
