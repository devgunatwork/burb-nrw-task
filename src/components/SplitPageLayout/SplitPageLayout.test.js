import React from 'react';
import SplitPageLayout from './SplitPageLayout';

describe('components', () => {
  describe('SplitPageLayout', () => {
    describe('UI', () => {
      it('should match the snapshot', () => {
        const wrapper = shallow(<SplitPageLayout />);

        expect(wrapper).toMatchSnapshot();
      });
    });
  });
});
