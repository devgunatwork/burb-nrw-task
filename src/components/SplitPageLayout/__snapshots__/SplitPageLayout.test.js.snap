// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components SplitPageLayout UI should match the snapshot 1`] = `
<WithStyles(ForwardRef(Grid))
  item={true}
  xs={12}
>
  <WithStyles(ForwardRef(Grid))
    className="root"
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      className="leftSide"
      item={true}
      xs={6}
    >
      <img
        alt="not available"
        src=""
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="rightSide"
      item={true}
      xs={6}
    />
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
